daemon/cmd/daemon.go

```diff
@@ -524,67 +520,19 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 	}
 
 	// Collect old CIDR identities
-	var oldNIDs []identity.NumericIdentity
-	var oldIngressIPs []netip.Prefix
 	if option.Config.RestoreState && !option.Config.DryMode {
-		if err := ipcachemap.IPCacheMap().DumpWithCallback(func(key bpf.MapKey, value bpf.MapValue) {
-			k := key.(*ipcachemap.Key)
-			v := value.(*ipcachemap.RemoteEndpointInfo)
-			nid := identity.NumericIdentity(v.SecurityIdentity)
-			if scope := nid.Scope(); scope == identity.IdentityScopeLocal ||
-				(scope == identity.IdentityScopeRemoteNode && option.Config.PolicyCIDRMatchesNodes()) {
-				d.restoredCIDRs = append(d.restoredCIDRs, k.Prefix())
-				oldNIDs = append(oldNIDs, nid)
-			} else if nid == identity.ReservedIdentityIngress && v.TunnelEndpoint.IsZero() {
-				prefix := k.Prefix()
-				oldIngressIPs = append(oldIngressIPs, prefix)
-				addr := prefix.Addr()
-				if addr.Is4() {
-					node.SetIngressIPv4(addr.AsSlice())
-				} else {
-					node.SetIngressIPv6(addr.AsSlice())
-				}
-			}
-		}); err != nil && !os.IsNotExist(err) {
-			log.WithError(err).Debug("Error dumping ipcache")
+		// this *must* becalled before initMaps(), which will clobber
+		// the "old" ipcache.
```

Comment:
nit on the space, and I think it's a little misleading to describe it as "clobbering"; the ambiguity of that wording could imply that the content of ipcache may be impacted, not just the pointer. `initMaps()` will change the open userspace reference to the ipcache so that it no longer refers to the instance that was used by the previous datapath + userspace and instead is a brand new fresh one to be used by the "new" instance of the datapath (which is about to be loaded).  Otherwise it's a good callout, this is a critical aspect which... unfortunately is only protected by this comment at this point. Maybe at some point we'll encode this constraint more reliably into the startup logic.

```suggestion
		// this *must* be called before initMaps(), which will hide
		// the "old" ipcache.
```

Reply:
Similar "clobber" nit on `restoreIPCache()`.

Reply:
the clobber's children have no... bpf maps?  (point taken)